client/src/controllers/ProgressController.ts

```diff
@@ -72,6 +72,13 @@ export class ProgressController extends Controller<HTMLButtonElement> {
     });
   }
 
+  deactivate() {
```

Comment:
I think it's useful to call this method inside `disconnect()`, so we don't duplicate the `clearTimeout()` code and we ensure that the `loadingValue` is set to `false` when the controller is disconnected (so if we reconnect it, it won't start with the "loading" value).